src/aspire/basis/basis.py

```diff
@@ -75,6 +75,18 @@ def norms(self):
         """
         raise NotImplementedError("subclasses must implement this")
 
+    def _coefficient_dtype(self):
```

Comment:
why do we need this?

Reply:
My thought was that maybe we would want to give `Basis` subclasses a function to determine what the output dtype would be. But we can just set it in the `__init__` and not do this.

Reply:
they can just set it.  If you are worried about the abc you can use abstractproperty in the base class.